### PR TITLE
Change canned response fields to be non-localised.  Fixes #1858

### LINKED ISSUE #1858: canned reponse editor/add new form is broken
The canned response edit page, e.g. https://addons-dev.allizom.org/en-US/admin/models/editors/cannedresponse/43/ is supposed to have fields where you can edit the name and response.  The new response form is similarly broken meaning we can't add any new ones either.

@jvillalobos

### LINKED ISSUE #1885: Unlisted addon fails to pass validation with an unexpected error
I've been able to use the automated validation/signing process to release my unlisted add-on until very recently. Whatever is in the most recent version of my addon breaks the validation process, I am unsure why.

Most recent version (named as .zip but it's my .xpi file from jpm): [firefox-at-getguru.com-1.1.24.2.zip](https://github.com/mozilla/addons-server/files/167247/firefox-at-getguru.com-1.1.24.2.zip)

The previous version (1.0.19.6) of the add on can be uploaded and revalidated just fine, so there is something different in the latest version that doesnt work at all.

Please attach a copy of your add-on, and fill in the following information:

* Whether this error occurred during the submission of a new add-on, a new version of an existing add-on, a new file in an existing version, or an upload to the standalone validator.

A new version of an exi

sting add-on

* Whether your add-on is listed or unlisted.

Unlisted

* The URL where you submitted the add-on.

https://addons.mozilla.org/en-US/developers/addon/guru/versions#version-upload

* The approximate time you submitted the add-on, with timezone.

March 9th, 2016 at 9:32pm, Eastern Standard Time

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #1858

### CODE REVIEW

File: src/olympia/editors/models.py [hunk 1]
```diff
@@ -29,8 +29,8 @@
 
 class CannedResponse(ModelBase):
 
-    name = TranslatedField()
-    response = TranslatedField(short=False)
+    name = models.CharField(max_length=255)
+    response = models.TextField()
```
Comment (diox) [inline src/olympia/editors/models.py]: If there are no translations left in this model you need to remove the `pre_save` signal connect below the model declaration as well.

Comment (eviljeff) [inline src/olympia/editors/models.py]: Done.  Is removing those lines enough or is something defined in translations also?

Comment (diox) [inline src/olympia/editors/models.py]: That's enough.

File: src/olympia/migrations/871-untranslate-canned-responses.sql [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/* Add the new columns and drop the constraints so we can clear the translations.
+   Local DB does have foreign key constraints, but AMO prod doesn't, so commenting
+   out those lines.*/
```
Comment (EnTeQuAk) [inline src/olympia/migrations/871-untranslate-canned-responses.sql]: sadpanda, :+1: for the comment though

**Comment by EnTeQuAk** [comment]
migration seems to work, code looks good.

I'm a bit torn regarding the migration and commenting out stuff for local-development though. This would fail on all new installations then right? Maybe check if foreign keys exist first or so? http://stackoverflow.com/questions/17161496/drop-foreign-key-only-if-it-exists/31121643#31121643 I am absolutely no mysql-expert though.

**Comment by eviljeff** [comment]
It wouldn't fail on _new _installations (because migrations are bypassed) but would on all existing local ones.  The stack overflow link shows its definitely possible to do something, but the foreign key name looks generated so would it be the same on every local install anyway?

File: src/olympia/migrations/871-untranslate-canned-responses.sql (hunk 2 above)
Comment (diox) [inline src/olympia/migrations/871-untranslate-canned-responses.sql]: I have those locally:

```
  CONSTRAINT `name_refs_id_8865ac11` FOREIGN KEY (`name`) REFERENCES `translations` (`id`),
  CONSTRAINT `response_refs_id_8865ac11` FOREIGN KEY (`response`) REFERENCES `translations` (`id`)
```

Pretty weird that they don't exist on dev/stage/prod.

**Comment by EnTeQuAk** [comment]
> It wouldn't fail on new installations (because migrations are bypassed)
Ah right, always forgetting that.

> but would on all existing local ones. The stack overflow link shows its definitely possible to do something, but the foreign key name looks generated so would it be the same on every local install anyway?

The names are mostly a md5 hash over the name and maybe some arguments (https://github.com/django/django/blob/stable/1.7.x/django/db/backends/creation.py#L59 if I am correct but it's something like that) - e.g if I run `manage.py sql` the generated column name looks the same as in your migration.

**Comment by diox** [comment]
New plan:
- Have @jasonthomas create the constraints (with the right names) on dev/stage/prod
- Include the `DROP FOREIGN KEY` statement in the migration.

**Comment by eviljeff** [comment]
migration updated.
To add the [constraints](https://pastebin.mozilla.org/8869847)

**Comment by diox** [comment]
r+ with a note in the push etherpad to add the constraints to stage / prod *before* deploying.

**Pull request merged**